src/index.ts

```diff
@@ -650,6 +650,40 @@ export function mount(forgoNode: ForgoNode, parentElement: HTMLElement | null) {
   }
 }
 
+interface HydrateOptions {
+  root?: string | (() => HTMLElement) | HTMLElement;
+}
+
+/*
+  Mount will hydrate the DOM as a child of the specified container element.
+*/
+export function hydrate(
+  forgoNode: ForgoNode,
+  parentElement: HTMLElement | null,
+  options?: HydrateOptions
+) {
+  const opts = options || {};
+  if (parentElement) {
+    let root;
+    if (typeof opts.root === "function") {
+      root = opts.root();
+    } else if (typeof opts.root === "object") {
+      root = opts.root;
+    } else {
+      root = parentElement.querySelector(opts.root || "[forgo-root]");
```

Comment:
might want to default to selecting first child instead of the query selector default.

Reply:
updated